src/components/events/EditSignup/EditForm.tsx

```diff
@@ -0,0 +1,178 @@
+import { useNavigate } from "@gatsbyjs/reach-router"
+import {
+  useEditSignupContext,
+  useUpdateSignup,
+} from "@tietokilta/ilmomasiina-components"
+import { Signup } from "@tietokilta/ilmomasiina-models"
+import { Field, Formik, FormikHelpers } from "formik"
+import { Link } from "gatsby"
+import { useCallback, useState } from "react"
+import { toast } from "react-toastify"
+import { H1, P } from "../../typography"
+import DeleteSignup from "./DeleteSignup"
+import FieldRow from "./FieldRow"
+import { Button, CheckBox, TextInput } from "../inputs"
+import QuestionFields from "./QuestionFields"
+import SignupStatus from "./SignupStatus"
+import { useEventsPaths } from "../utils"
+
+const EditForm = () => {
+  const { event, signup, registrationClosed } = useEditSignupContext()
+  const isNew = signup!.confirmedAt === null
+  const updateSignup = useUpdateSignup()
+  const navigate = useNavigate()
+  const paths = useEventsPaths()
+
+  // TODO: actually use errors from API
+  const [submitError, setSubmitError] = useState(false)
+
+  const onSubmit = useCallback(
+    async (
+      answers: Signup.Update.Body,
+      { setSubmitting }: FormikHelpers<Signup.Update.Body>
+    ) => {
+      const action = isNew ? "Ilmoittautuminen" : "Muokkaus"
+      const progressToast = toast.loading(`${action} käynnissä`)
+
+      try {
+        await updateSignup(answers)
+
+        toast.update(progressToast, {
+          render: `${action} onnistui!`,
+          type: toast.TYPE.SUCCESS,
+          autoClose: 5000,
+          closeButton: true,
+          closeOnClick: true,
+          isLoading: false,
+        })
+        setSubmitError(false)
+        setSubmitting(false)
+        if (isNew) {
+          navigate(paths.eventDetails(event!.slug))
+        }
+      } catch (error) {
+        toast.update(progressToast, {
+          render: `${action} ei onnistunut. Tarkista, että kaikki pakolliset kentät on täytetty ja yritä uudestaan.`,
+          type: toast.TYPE.ERROR,
+          autoClose: 5000,
+          closeButton: true,
+          closeOnClick: true,
+          isLoading: false,
+        })
+        setSubmitError(true)
+        setSubmitting(false)
+      }
+    },
+    [event, isNew, updateSignup, navigate, paths]
+  )
+
+  return (
+    <Formik initialValues={signup! as Signup.Update.Body} onSubmit={onSubmit}>
```

Comment:
`if (!signup) return null` or some component indicating that something went wrong.